add await to the async isValidGitDirectory

icecream17 opened the pull request:
### Requirements for Contributing a Bug Fix

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix an existing bug. To contribute other changes, you must use a different template. You can see all templates at <https://github.com/atom/.github/tree/master/.github/PULL_REQUEST_TEMPLATE>.
* The pull request must update the test suite to demonstrate the changed functionality. For guidance, please see <https://flight-manual.atom.io/hacking-atom/sections/writing-specs/>.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see <https://github.com/atom/.github/tree/master/CONTRIBUTING.md#pull-requests>.

### Identify the Bug

Fixes #22577
<!--

Link to the issue describing the bug that you're fixing.

If there is not yet an issue for your bug, please open a new issue and then link to that issue in your pull request.
Note: In some cases, one person's "bug" is another person's "feature." If the pull request does not address an existing issue with the "bug" label, the maintainers have the final say on whether the current behavior is a bug.

-->

### Description of the Change

Pretty self explanatory

isValidGitDirectory is an async function, and so actually returns a Promise<boolean> which is always truthy

<!--

We must be able to understand the design of your change from this description. If we can't get a good idea of what the code will be doing from the description here, the pull request may be closed at the maintainers' discretion. Keep in mind that the maintainer reviewing this PR may not be familiar with or have worked with the code here recently, so please walk us through the concepts.

-->

### Alternate Designs

N/A
<!-- Explain what other alternates were considered and why the proposed version was selected -->

### Possible Drawbacks

N/A
<!-- What are the possible side-effects or negative impacts of the code change? -->

### Verification Process

N/A (Hopefully)
<!--

What process did you follow to verify that the change has not introduced any regressions? Describe the actions you performed (including buttons you clicked, text you typed, commands you ran, etc.), and describe the results you observed.

-->

### Release Notes

The findGitDirectory function now checks if the directory is valid
<!--

Please describe the changes in a single line that explains this improvement in
terms that a user can understand. This text will be used in Atom's release notes.

If this change is not user-facing or notable enough to be included in release notes
you may use the strings "Not applicable" or "N/A" here.

Examples:

- The GitHub package now allows you to add co-authors to commits.
- Fixed an issue where multiple cursors did not work in a file with a single line.
- Increased the performance of searching and replacing across a whole project.

-->

Commit:
add await to the async isValidGitDirectory

Since isValidGitDirectory is async, it actually returns Promise<boolean> which is always truthy

That's probably not what the code meant

icecream17 commented:
There might also be a bug on line 144 with non-awaited promises, but lgtm.com doesn't say anything about it, so maybe it's intentional??

Commit:
Merge branch 'atom:master' into patch-1

UziTech commented:
Looks like line 144 is just returning the promise instead of returning the awaited value.

It shouldn't make a difference but I feel like it should await the last call just to be consistent.

Commit:
await `.exists()` for consistency

See comment https://github.com/atom/atom/pull/22578#issuecomment-864708333

What I didn't realize was that line 144 was in an async return. So this commit shouldn't actually change any functionality.
Nevertheless, await it for consistency.
